lib/charms/mongodb_libs/v0/mongodb_provider.py

```diff
@@ -187,10 +220,35 @@ def _on_legacy_relation_joined(self, event):
             "replset": self.charm.app.name,
         }
 
-        # reactive charms set relation data on "the current unit"
+        # reactive charms set relation data on "the current unit" the reactive mongodb charm sets
+        # the relation data for all units, hence all units setting the relation data and not just
+        # the leader
         relation = self.model.get_relation(REL_NAME, event.relation.id)
         relation.data[self.charm.unit].update(updates)
 
+    def get_users_from_legacy_relations(self):
+        """Return usernames for legacy relation."""
+        relations = self.model.relations[LEGACY_REL_NAME]
+        logger.debug(relations)
+        return set([self._get_username_from_relation_id(relation.id) for relation in relations])
+
+    # TODO move to machine charm helpers
+    def _auth_enabled(self):
+        """Checks if mongod service is running with auth enabled."""
+        if not os.path.exists("/lib/systemd/system/mongod.service"):
+            return False
+
+        with open("/lib/systemd/system/mongod.service", "r") as mongodb_service_file:
+            mongodb_service = mongodb_service_file.readlines()
+
+        for index, line in enumerate(mongodb_service):
+            # ExecStart contains the line with the arguments to start mongod service.
+            if "ExecStart" in line:
+                if "--auth" in line:
+                    return True
```

Comment:
Am I right in thinking you don't use index here?

Reply:
you can also do:

`if "ExecStart" and "--auth" in line:`

Reply:
I guess `if "ExecStart" and "--auth" in line:` always evaluates `"ExecStart"` as True, because of the string content.

Reply:
why didn't i see this earlier 🤦

Reply:
Small change, `if "ExecStart" and "--auth" in line` will evaluate `ExecStart` as always true since it's a non empty string:
```
if "ExecStart" in line and "--auth" in line:
```
 or 
```
if any(i in line for i in ("ExecStart", "--auth"))
```

Reply:
Heh, race conditions on github comments :)

Reply:
opting for `if "ExecStart" in line and "--auth" in line:` thanks for the help :)